main/src/library/Iterator.flix

```diff
@@ -589,199 +517,169 @@ namespace Iterator {
     ///
     /// The original iterator `iter` should *not* be reused.
     ///
-    @Lazy
-    pub def takeL(n: Int32, iter: Iterator[a, r]): Iterator[a, r] \ Write(r) =
-        let r = Scoped.regionOf(iter);
-        if (n < 0)
-            new Iterator(r)
-        else
-            let Iterator(done, next) = iter;
-            let i = ref 0 @ r;
-            let done1 = () -> done() or (deref i) >= n;
-            let next1 = () -> {
-                let x = next();
-                i := deref i + 1;
-                x
-            };
-            Iterator(done1, next1)
+    /// SPT TODO - check effect of `iter`
```

Comment:
Something with the effect?

Reply:
I think this was `drop` which I changed the signature to "region access" is in the effect set of the return type.